src/audio/kpb.c

```diff
@@ -595,12 +603,20 @@ static int kpb_copy(struct comp_dev *dev)
 		}
 
 		copy_bytes = MIN(sink->free, source->avail);
+		if (!copy_bytes) {
+			trace_kpb_error("kpb_copy() error: nothing to copy "
+					"sink->free %d sorce->avail %d",
+					sink->free, source->avail);
+			ret = PPL_STATUS_PATH_STOP;
+			goto out;
+		}
+
 		kpb_copy_samples(sink, source, copy_bytes, sample_width);
 
 		comp_update_buffer_produce(sink, copy_bytes);
 		comp_update_buffer_consume(source, copy_bytes);
 
-		ret = 0;
+		ret = PPL_STATUS_PATH_STOP;
```

Comment:
Is it intentional change?

Reply:
Yes, during host copy we no longer need selector and detector components.